letsencrypt/auth_handler.py

```diff
@@ -153,13 +153,18 @@ def _send_responses(self, achalls, resps, chall_update):
         """
         active_achalls = []
         for achall, resp in itertools.izip(achalls, resps):
+            # XXX: make sure that all achalls, including those
+            # corresponding to None or False returned from
+            # Authenticator are removed from the queue and thus avoid
+            # infinite loop
+            active_achalls.append(achall)
```

Comment:
I need to relook at the code.  It was thought that there might be authenticators that can only handle so many challenges at a time.  Easy example: A server that can only solve 1 DVSNI challenge at a time due to SNI constraints or lack of SNI support.  There are many other more complicated examples where you might not be able to provision all resources at once.  (I had intended for the API to be adaptable to future challenge types and unknown plugin constraints)

There used to be a countdown mechanism where it was guaranteed at least one challenge was being solved per round.  I am not sure if that has been stripped out.... but that was the intent.  I will take a closer look.